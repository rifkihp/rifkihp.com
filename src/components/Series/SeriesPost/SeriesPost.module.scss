@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.series-post {
  margin: 20px (-$layout-mobile-padding);
  background-color: $color-gray-bg;

  &-content {
    padding: 15px $layout-mobile-padding;
  }

  img {
    margin: 0 auto;
    border-radius: 3px;
  }

  h3 {
    margin: 10px 0 5px 0;
  }

  p {
    margin: 0;
  }
}

@include breakpoint-sm {
  .series-post {
    max-width: $layout-post-width;
    margin: 25px auto;
    border-radius: 5px;

    &-content {
      padding: 20px 25px;
    }
  }
}

@include breakpoint-md {
  .series-post {
    margin: 30px auto;

    &-content {
      padding: 25px 35px;
    }

    p {
      margin: 5px 0 0 0;
    }
  }
}
