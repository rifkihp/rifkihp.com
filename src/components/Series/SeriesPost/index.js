// @flow
export { default } from './SeriesPost';
