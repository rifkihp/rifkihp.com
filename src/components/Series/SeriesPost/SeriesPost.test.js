import React from 'react';
import renderer from 'react-test-renderer';

import SeriesPost from './SeriesPost';

describe('SeriesPost', () => {
  const props = {
    dateFormatted: 'March 14, 2019',
    description: 'test-description',
    img: 'test-img',
    n: 1,
    slug: 'test-slug',
    title: 'test-title',
  };

  it('renders correctly', () => {
    const tree = renderer.create(<SeriesPost {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
