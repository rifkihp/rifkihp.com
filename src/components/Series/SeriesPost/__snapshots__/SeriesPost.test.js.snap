// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeriesPost renders correctly 1`] = `
<div
  className="series-post"
>
  <div
    className="series-post-content"
  >
    <Link
      to="test-slug"
    >
      <img
        alt="test-title"
        loading="lazy"
        src="test-img"
      />
    </Link>
    <div>
      <h3>
        <Link
          to="test-slug"
        >
          1. test-title
        </Link>
      </h3>
      <p
        className="content-date"
      >
        <time>
          March 14, 2019
        </time>
      </p>
      <p>
        test-description
      </p>
    </div>
  </div>
</div>
`;
